internal/log/log.go

```diff
@@ -64,45 +66,56 @@ func CreateLogger(
 	pidPath string,
 	removePIDfunc removePIDfunc,
 	foreground bool,
-	bootstrapLogger zap.Logger,
-) (zap.Logger, error) {
+	bootstrapLogger *Logger,
+) (*Logger, error) {
 
-	output := zap.Output(os.Stdout)
+	output := []string{"stderr"}
+	if c.StdOut || foreground {
+		output = []string{"stdout"}
+	}
 	if c.LogSink != "" {
-		sink, err := os.OpenFile(c.LogSink, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0644)
-		if err != nil {
-			return nil, err
-		}
 		bootstrapLogger.Info("Log file path provided", zap.String("path", c.LogSink))
-		if c.StdOut || foreground {
-			output = zap.Output(zap.MultiWriteSyncer(os.Stdout, sink))
-		} else {
-			output = zap.Output(sink)
-		}
+		output = append(output, c.LogSink)
+	}
+
+	initialFields := map[string]interface{}{
+		"service_name": service,
+		"hostname":     hostname,
+		"PID":          os.Getpid(),
+	}
+
+	var level zapcore.Level
+	if err := level.Set(c.Level); err != nil {
+		bootstrapLogger.Error("Log level provided", zap.Error(err))
+	}
+
+	config := zap.Config{
+		Level:            zap.NewAtomicLevelAt(level),
+		Development:      false,
+		DisableCaller:    true,
+		EncoderConfig:    zap.NewProductionEncoderConfig(),
+		Encoding:         "json",
+		ErrorOutputPaths: []string{"stdout"},
+		OutputPaths:      output,
+		InitialFields:    initialFields,
 	}
 
-	fields := zap.Fields(
-		zap.String("service_name", service),
-		zap.String("hostname", hostname),
-		zap.Int("PID", os.Getpid()),
-	)
-
-	logger := PIDRemoverLogger{
-		Logger: zap.New(
-			zap.NewJSONEncoder(),
-			c.Level,
-			fields,
-			output),
+	l, err := config.Build()
+	if err != nil {
+		return nil, errors.New("failed to create logger")
```

Comment:
BTW, the err message was not hidden but displayed by the calling function:

`logger, err := log.CreateLogger(&gl.App.Logging, d.ArachneService, osHostname,
		gl.App.PIDPath, util.RemovePID, *(gl.CLI.Foreground), bootstrapLogger)
	if err != nil {
		bootstrapLogger.Error("unable to initialize Arachne Logger", zap.Error(err))
		os.Exit(1)
	}`